setup

```diff
@@ -35,10 +35,39 @@ CMDLINE_ARGUMENT_LIST=(
   "skip-systemctl"
 )
 
+logger() {
+    local log_level="$1"
+    local message="$2"
+    {
+        local NC='\033[0m' # No Color
+        local log_col=$NC
+        case "$log_level" in
+            INFO)
+                log_col='\033[0;36m' #BLUE
+            ;;
+            WARNING)
+                log_col='\033[1;33m' #YELLOW
+            ;;
+            ERROR)
+                log_col='\033[0;31m' #RED
+                ;;
+            SUCCESS)
+                log_col='\033[0;32m' #GREEN
+            ;;
+            *)
+                log_col='\033[0;31m' #RED
+                echo -e "[\033[1;31mERROR\033[0m] Invalid log level: $log_level" >&2
+                return 1
+            ;;
+        esac
+    }
+  echo -e "[$log_col${log_level}${NC}] $message"
```

Comment:
Consider using exit code 
logger "ERROR" "Exiting... '$file' not found. Try reinstall the QM package before continuing." EXIT